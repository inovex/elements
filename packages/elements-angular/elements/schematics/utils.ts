import { virtualFs, workspaces } from '@angular-devkit/core';
import { SchematicsException, Tree } from '@angular-devkit/schematics';

const CONFIG_PATH = 'angular.json';

function createHost(tree: Tree): workspaces.WorkspaceHost {
  return {
    async readFile(path: string): Promise<string> {
      const data = tree.read(path);
      if (!data) {
        throw new SchematicsException('File not found.');
      }
      return virtualFs.fileBufferToString(data);
    },
    async writeFile(path: string, data: string): Promise<void> {
      return tree.overwrite(path, data);
    },
    async isDirectory(path: string): Promise<boolean> {
      return !tree.exists(path) && tree.getDir(path).subfiles.length > 0;
    },
    async isFile(path: string): Promise<boolean> {
      return tree.exists(path);
    },
  };
}

export function readConfig(host: Tree) {
  const sourceText = host.read(CONFIG_PATH)?.toString('utf-8') || '';
  return JSON.parse(sourceText);
}

export function writeConfig(host: Tree, config: JSON) {
  host.overwrite(CONFIG_PATH, JSON.stringify(config, null, 2));
}

export function getAngularAppConfig(
  config: any,
  projectName: string
): any | never {
  if (!config.projects.hasOwnProperty(projectName)) {
    throw new SchematicsException(`Could not find project: ${projectName}`);
  }

  return config.projects[projectName];
}

export async function getProjectName(tree: Tree) {
  const host = createHost(tree);
  const { workspace } = await workspaces.readWorkspace('/', host);
  return workspace.extensions.defaultProject as string;
}
