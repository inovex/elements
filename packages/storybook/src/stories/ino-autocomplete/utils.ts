import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
