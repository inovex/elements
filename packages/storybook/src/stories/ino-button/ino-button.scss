.story-button {

  ino-button {
    margin-bottom: 10px;
  }

  .button-row {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    margin-bottom: 10px;
  }

  .reset-form {
    display: flex;

    .reset-input {
      flex: 1;
      margin: 0 12px 12px 0;
    }
  }

  .white-button {
    background-color: blue;
    height: 100%;
    width: min-content;
    padding: 5px;
  }
}
