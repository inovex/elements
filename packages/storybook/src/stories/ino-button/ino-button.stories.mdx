import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-button/readme.md";

<Meta title="Buttons/ino-button/Documentation" />

<Description markdown={readme} />
