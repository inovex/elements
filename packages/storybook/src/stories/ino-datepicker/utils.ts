import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
