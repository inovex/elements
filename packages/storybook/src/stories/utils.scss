/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
