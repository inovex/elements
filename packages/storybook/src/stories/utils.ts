import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
