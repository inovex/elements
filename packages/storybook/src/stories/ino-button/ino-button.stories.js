import { boolean, select, text } from '@storybook/addon-knobs';

import componentReadme from '_local-elements/src/components/ino-button/readme.md';
import withStencilReadme from '_local-storybookcore/with-stencil-readme';
import ICONS from '_local-elements/src/components/ino-icon/icons';

import './ino-button.scss';

export default {
  title: 'Buttons/<ino-button>',
  parameters: {
    actions: {
      handles: [
        'click .customizable-button',
        'reset .reset-form',
        'submit .submit-form',
      ],
    },
  },
  decorators: [withStencilReadme(componentReadme)],
};

export const DefaultUsage = () => /* html */ `
    <div class="story-button">
      <style>
          ino-button.customizable-button {
            --button-color-primary: ${text(
              '--button-color-primary',
              '#3d40f5',
              'Custom Properties'
            )};
            --button-color-primary-light: ${text(
              '--button-color-primary-light',
              '#5d60f7',
              'Custom Properties'
            )};
            --button-color-primary-dark: ${text(
              '--button-color-primary-dark',
              '#0d10f3',
              'Custom Properties'
            )};
            --button-color-primary-contrast: ${text(
              '--button-color-primary-contrast',
              '#fff',
              'Custom Properties'
            )};
            --button-color-primary-shadow-light: ${text(
              '--button-color-primary-shadow-light',
              'rgba(61, 214, 245, 0.14)',
              'Custom Properties'
            )};
            --button-color-primary-shadow-dark: ${text(
              '--button-color-primary-shadow-light',
              'rgba(61, 64, 245, 0.3)',
              'Custom Properties'
            )};
            --button-color-secondary: ${text(
              '--button-color-secondary',
              '#9ccd00',
              'Custom Properties'
            )};
            --button-color-secondary-light: ${text(
              '--button-color-secondary-light',
              '#bbe962',
              'Custom Properties'
            )};
            --button-color-secondary-dark: ${text(
              '--button-color-secondary-dark',
              '#93bf00',
              'Custom Properties'
            )};
            --button-color-secondary-contrast: ${text(
              '--button-color-secondary-contrast',
              '#fff',
              'Custom Properties'
            )};
            --button-color-secondary-shadow-light: ${text(
              '--button-color-secondary-shadow-light',
              'rgba(156, 205, 0, 0.14)',
              'Custom Properties'
            )};
            --button-color-secondary-shadow-dark: ${text(
              '--button-color-secondary-shadow-light',
              'rgba(181, 230, 23, 0.30)',
              'Custom Properties'
            )};
          }
      </style>
      <ino-button
        class="customizable-button"
        ino-color-scheme="${select(
          'ino-color-scheme',
          ['primary', 'secondary', 'white', 'grey'],
          'primary',
          'General'
        )}"
        ino-fill="${select(
          'ino-fill',
          ['solid', 'outline', 'inverse'],
          'solid',
          'General'
        )}"
        ino-icon-leading="${boolean('ino-icon-leading', false, 'Icon')}"
        ino-icon-trailing="${boolean('ino-icon-trailing', false, 'Icon')}"
        ino-dense="${boolean('ino-dense', false, 'General')}"
        ino-full-width="${boolean('ino-full-width', false, 'General')}"
        disabled="${boolean('disabled', false, 'General')}"
        ino-loading="${boolean('ino-loading', false, 'General')}"
        ino-edge-mirrored="${boolean('ino-edge-mirrored', false, 'General')}"
      >
        <ino-icon slot="ino-icon-leading" ino-icon="${select(
          'ino-icon-leading-slot',
          ICONS,
          'Add',
          'Icon'
        )}"></ino-icon>
        ${text('<slot />', 'Customizable button', 'General')}
        <ino-icon slot="ino-icon-trailing" ino-icon="${select(
          'ino-icon-trailing-slot',
          ICONS,
          'Add',
          'Icon'
        )}"></ino-icon>
      </ino-button>

      <h4>Variations</h4>
      <div class="button-row">
        <ino-button ino-fill="solid">Solid Primary</ino-button>
        <ino-button ino-fill="outline">Outline Primary</ino-button>
        <ino-button ino-fill="inverse">Inverse Primary</ino-button>
      </div>
      <div class="button-row">
        <ino-button ino-fill="solid" ino-color-scheme="secondary">Solid Secondary </ino-button>
        <ino-button ino-fill="outline" ino-color-scheme="secondary">Outline Secondary</ino-button>
        <ino-button ino-fill="inverse" ino-color-scheme="secondary">Inverse Secondary</ino-button>
      </div>
      <div class="button-row">
        <ino-button ino-fill="outline" ino-color-scheme="grey">Outline Grey</ino-button>
        <div style="background-color: blue; height: 100%; padding: 5px;">
            <ino-button style="width: 50%; margin: 0 auto;" ino-fill="outline" ino-color-scheme="white">Outline White</ino-button>
        </div>
      </div>

      <h4>With icons</h4>
      <ino-button ino-icon-leading>
        <ino-icon ino-icon="info" slot="ino-icon-leading"></ino-icon>
        Button Icon left
      </ino-button>
      <ino-button ino-icon-trailing>
        <ino-icon ino-icon="info" slot="ino-icon-trailing"></ino-icon>
        Button Icon right
      </ino-button>

      <h4>States</h4>
      <div class="button-row">
        <ino-button disabled>Disabled</ino-button>
        <ino-button disabled ino-fill="outline">Disabled outlined</ino-button>
        <ino-button ino-edge-mirrored="true">With mirrored edge</ino-button>
        <ino-button ino-dense="true">Dense</ino-button>
        <ino-button ino-loading="true">Loading button</ino-button>
        <ino-button ino-loading="true" ino-fill="outline">Loading button</ino-button>
       </div>
       <ino-button ino-full-width>Full Width Button</ino-button>
    </div>
  `;

export const Forms = () => /*html*/ `
    <script>
      const form = querySelector('form');
      form.addEventListener('submit', e => e.preventDefault());
    </script>

    <div class="story-button">
      <h4>Form submit</h4>
      <form class="submit-form" onsubmit="return false;">
        <ino-button type="submit">Button to submit parent form</ino-button>
        <ino-button type="submit" disabled>Disabled button does not submit form</ino-button>
      </form>

      <h4>Form reset</h4>
      <form class="reset-form">
        <ino-input class="reset-input" placeholder="Input in reset form" value="Value"></ino-input>
        <ino-button type="reset">Reset form values on the left</ino-button>
      </form>

      <h4>Button that submits external form with form attribute</h4>
      <form class="submit-form" id="form1" onsubmit="return false;">
      <ino-button type="submit" form="form1">Button with form attribute</ino-button>
      </form>
    </div>
  `;
