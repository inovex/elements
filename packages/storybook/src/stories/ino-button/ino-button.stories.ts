import { Components } from '@inovex.de/elements';
import { Story } from '@storybook/web-components';
import { html } from 'lit-html';
import { defaultDecorator } from '../utils';

import './ino-button.scss';

export default {
  title: 'Buttons/<ino-button>',
  component: 'ino-button',
  decorators: [defaultDecorator],
  parameters: {
    actions: {
      handles: [
        'click .customizable-button',
        'reset .reset-form',
        'submit .submit-form',
      ],
    },
  },
};

export const Playground: Story<Components.InoButton> = (
  args
) => html` <ino-button
  class="customizable-button"
  color-scheme="${args.colorScheme}"
  fill="${args.fill}"
  dense="${args.dense}"
  disabled="${args.disabled}"
  loading="${args.loading}"
  edge-mirrored="${args.edgeMirrored}"
>
  Label
</ino-button>`;

Playground.args = {
  colorScheme: 'primary',
  fill: 'solid',
  dense: false,
  disabled: false,
  loading: false,
  edgeMirrored: false,
};

Playground.argTypes = {
  colorScheme: {
    control: {
      type: 'select',
    },
    options: ['primary', 'secondary', 'grey', 'white'],
  },
  fill: {
    control: {
      type: 'select',
    },
    options: ['solid', 'outline', 'inverse'],
  },
};

export const Others = () => html`
  <div class="story-button">
    <h4>Variations</h4>
    <div class="button-row">
      <ino-button fill="solid">Solid Primary</ino-button>
      <ino-button fill="outline">Outline Primary</ino-button>
      <ino-button fill="inverse">Inverse Primary</ino-button>
    </div>
    <div class="button-row">
      <ino-button fill="solid" color-scheme="secondary"
        >Solid Secondary</ino-button
      >
      <ino-button fill="outline" color-scheme="secondary"
        >Outline Secondary</ino-button
      >
      <ino-button fill="inverse" color-scheme="secondary"
        >Inverse Secondary</ino-button
      >
    </div>
    <div class="button-row">
      <ino-button fill="outline" color-scheme="grey">Outline Grey</ino-button>
      <div class="white-button">
        <ino-button fill="outline" color-scheme="white"
          >Outline White</ino-button
        >
      </div>
    </div>

    <h4>With icons</h4>
    <div class="button-row">
      <ino-button>
        <ino-icon icon="info" slot="icon-leading"></ino-icon>
        Button Icon left
      </ino-button>
      <ino-button>
        <ino-icon icon="info" slot="icon-trailing"></ino-icon>
        Button Icon right
      </ino-button>
    </div>
    <h4>States</h4>
    <div class="button-row">
      <ino-button disabled>Disabled</ino-button>
      <ino-button disabled fill="outline">Disabled outlined</ino-button>
      <ino-button edge-mirrored="true">With mirrored edge</ino-button>
      <ino-button dense="true">Dense</ino-button>
      <ino-button loading="true">Loading button</ino-button>
      <ino-button loading="true" fill="outline">Loading button</ino-button>
    </div>
  </div>
`;

export const Forms = () => html`
  <script>
    const form = querySelector('form');
    form.addEventListener('submit', (e) => e.preventDefault());
  </script>

  <div class="story-button">
    <h4>Form submit</h4>
    <form class="submit-form" onsubmit="return false;">
      <ino-button type="submit">Button to submit parent form</ino-button>
      <ino-button type="submit" disabled
        >Disabled button does not submit form
      </ino-button>
    </form>

    <h4>Form reset</h4>
    <form class="reset-form">
      <ino-input
        class="reset-input"
        placeholder="Input in reset form"
        value="Value"
      ></ino-input>
      <ino-button type="reset">Reset form values on the left</ino-button>
    </form>

    <h4>Button that submits external form with form attribute</h4>
    <form class="submit-form" id="form1" onsubmit="return false;">
      <ino-button type="submit" form="form1"
        >Button with form attribute
      </ino-button>
    </form>
  </div>
`;
