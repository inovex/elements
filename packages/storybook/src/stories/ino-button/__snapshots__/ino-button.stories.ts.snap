// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons/<ino-button> Dense smoke-test 1`] = `
<div id="root-inner">
  <div class="story-button">
    <ino-button variant="filled"
                disabled="false"
                dense="true"
                loading="false"
                class="hydrated"
    >
      <button class="button button__variant--filled button--dense"
              type="button"
      >
        <span class="button__label">
          Label
        </span>
      </button>
    </ino-button>
  </div>
</div>
`;

exports[`Buttons/<ino-button> Disabled smoke-test 1`] = `
<div id="root-inner">
  <div class="story-button">
    <ino-button variant="filled"
                disabled="true"
                dense="false"
                loading="false"
                class="hydrated"
    >
      <button class="button button__variant--filled"
              disabled
              type="button"
      >
        <span class="button__label">
          Label
        </span>
      </button>
    </ino-button>
  </div>
</div>
`;

exports[`Buttons/<ino-button> Filled smoke-test 1`] = `
<div id="root-inner">
  <div class="story-button">
    <ino-button variant="filled"
                disabled="false"
                dense="false"
                loading="false"
                class="hydrated"
    >
      <button class="button button__variant--filled"
              type="button"
      >
        <span class="button__label">
          Label
        </span>
      </button>
    </ino-button>
  </div>
</div>
`;

exports[`Buttons/<ino-button> LeadingAndTrailingIcon smoke-test 1`] = `
<div id="root-inner">
  <div class="story-button">
    <div class="container">
      <ino-button class="hydrated">
        <button class="button button__variant--filled button__icon--leading"
                type="button"
        >
          <span class="icon__wrapper">
            <ino-icon icon="info"
                      slot="icon-leading"
                      class="hydrated ino-icon--color-scheme-primary"
            >
            </ino-icon>
          </span>
          <span class="button__label">
            Label
          </span>
        </button>
      </ino-button>
      <ino-button class="hydrated">
        <button class="button button__variant--filled button__icon--trailing"
                type="button"
        >
          <span class="button__label">
            Label
          </span>
          <span class="icon__wrapper">
            <ino-icon icon="info"
                      slot="icon-trailing"
                      class="hydrated ino-icon--color-scheme-primary"
            >
            </ino-icon>
          </span>
        </button>
      </ino-button>
    </div>
  </div>
</div>
`;

exports[`Buttons/<ino-button> Loading smoke-test 1`] = `
<div id="root-inner">
  <div class="story-button">
    <ino-button variant="filled"
                disabled="false"
                dense="false"
                loading
                class="ino-button--loading hydrated"
    >
      <button class="button button__variant--filled"
              type="button"
      >
        <span class="button__label button__label--hide">
          Label
        </span>
        <ino-spinner class="ino-spinner--type-circle ino-spinner--color-scheme-primary hydrated">
          <div class="ino-spinner__composer"
               style="width: 20px; height: 20px;"
          >
            <div class="ino-spinner__circle-1">
            </div>
            <div class="ino-spinner__circle-2">
            </div>
            <div class="ino-spinner__circle-3">
            </div>
            <div class="ino-spinner__circle-4">
            </div>
            <div class="ino-spinner__circle-5">
            </div>
            <div class="ino-spinner__circle-6">
            </div>
            <div class="ino-spinner__circle-7">
            </div>
            <div class="ino-spinner__circle-8">
            </div>
            <div class="ino-spinner__circle-9">
            </div>
            <div class="ino-spinner__circle-10">
            </div>
            <div class="ino-spinner__circle-11">
            </div>
            <div class="ino-spinner__circle-12">
            </div>
          </div>
        </ino-spinner>
      </button>
    </ino-button>
  </div>
</div>
`;

exports[`Buttons/<ino-button> Outlined smoke-test 1`] = `
<div id="root-inner">
  <div class="story-button">
    <ino-button variant="outlined"
                disabled="false"
                dense="false"
                loading="false"
                class="hydrated"
    >
      <button class="button button__variant--outlined"
              type="button"
      >
        <span class="button__label">
          Label
        </span>
      </button>
    </ino-button>
  </div>
</div>
`;

exports[`Buttons/<ino-button> Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-button">
    <ino-button variant="filled"
                disabled="false"
                dense="false"
                loading="false"
                class="hydrated"
    >
      <button class="button button__variant--filled"
              type="button"
      >
        <span class="button__label">
          Label
        </span>
      </button>
    </ino-button>
  </div>
</div>
`;

exports[`Buttons/<ino-button> Text smoke-test 1`] = `
<div id="root-inner">
  <div class="story-button">
    <ino-button variant="text"
                disabled="false"
                dense="false"
                loading="false"
                class="hydrated"
    >
      <button class="button button__variant--text"
              type="button"
      >
        <span class="button__label">
          Label
        </span>
      </button>
    </ino-button>
  </div>
</div>
`;
